src/2-Circles/CircleDisplay.tsx

```diff
@@ -0,0 +1,490 @@
+import { TabNavigationProps, CircleState, StackNavigationProps } from '../TypesAndInterfaces/custom-types';
+import theme, { COLORS, FONTS, FONT_SIZES } from '../theme';
+
+import { BottomTabBarHeightCallbackContext } from '@react-navigation/bottom-tabs';
+import { CircleStatusEnum } from '../TypesAndInterfaces/config-sync/input-config-sync/circle-field-config';
+import { useAppDispatch, useAppSelector } from '../TypesAndInterfaces/hooks';
+import { RootState, addCircle, removeCircle } from '../redux-store';
+import { AnnouncementTouchable, CircleTouchable, EventTouchable, Input_Field, PrayerRequestTouchable, Raised_Button } from '../widgets';
+
+// TODO: Make circle lookup page AS modal
+// Also think about a dynamic lookup system for associating circles, users, prayer requests, etc
+
+// Look into useLayoutEffect
+
+const CircleDisplay = ({navigation, route}:StackNavigationProps, status: number):JSX.Element => {
+    const dispatch = useAppDispatch();
+    const jwt = useAppSelector((state: RootState) => state.account.jwt);
+    const userID = useAppSelector((state: RootState) => state.account.userID);
+    const userCircles = useAppSelector((state:RootState) => state.account.userProfile.circleList);
+    
+    const routeProps = route.params;
+    const circleProps = routeProps.CircleProps as unknown as CircleListItem;
+
+    const RequestAccountHeader = {
+      headers: {
+        "jwt": jwt, 
+        "userID": userID,
+      }
+    }
+
+    const defaultState = {
+        circleData: {} as CircleResponse,
+        circleDataRequest: {} as CircleListItem
+    } as CircleState;
+
+    var renderState = {
+        ...defaultState
+    }
+
+    const [leaveCircleModalVisible, setLeaveCircleModalVisible] = useState(false) // show circle leave warning
+    const [circleInfoModalVisible, setCircleInfoModalVisible] = useState(false); // show circle leader, circle name, and circle settings
+    const [currCircleState, setCurrCircleState] = useState(defaultState); // holds all current circle data
+    const [announcementsData, setAnnouncementsData] = useState([] as CircleAnnouncementListItem[]);
+    const [prayerRequestsData, setPrayerRequestsData] = useState([] as PrayerRequestListItem[]);
+    const [eventsData, setEventsData] = useState([] as CircleEventListItem[]);
+    const [dataFetchComplete, setDataFetchComplete] = useState(false); // toggles wait screen 
+
+    const renderEvents = ():JSX.Element[] => {
+        var eventModals:JSX.Element[] = [];
+        if (eventsData !== undefined) {
+            eventsData.forEach((event:CircleEventListItem, index) => {
+                eventModals.push(
+                    <EventTouchable
+                        key={index}
+                        circleEvent={event}
+                        onPress={() => console.log("Event callback")}
+                    />
+                )
+            });
+        }
+        return eventModals;
+    }
```

Comment:
If you're not doing any additional logic; I think a `.map` is cleaner.  I often propagate the JSX elements in the `return()`  directly; but I can see the usefulness of separating out the lists for readability.
```
    const renderEvents = ():JSX.Element[] => 
        (eventsData || []).map((event:CircleEventListItem, index:number) =>
                <EventTouchable
                    key={index}
                    circleEvent={event}
                    onPress={() => console.log("Event callback")}
                />
            );
      
```
Then in `return`:
```
<ScrollView horizontal={true} contentContainerStyle={styles.eventScroll}>
       {...renderEvents()}
</ScrollView>
```